cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -228,19 +236,24 @@ impl YaccParser {
                 i = self.parse_ws(j, false)?;
                 let num_newlines = self.num_newlines;
                 if self.ast.avoid_insert.is_none() {
-                    self.ast.avoid_insert = Some(HashSet::new());
+                    self.ast.avoid_insert = Some(HashMap::new());
                 }
                 while j < self.src.len() && self.num_newlines == num_newlines {
                     let (j, n, span) = self.parse_token(i)?;
                     if self.ast.tokens.insert(n.clone()) {
                         self.ast.spans.push(span);
                     }
-                    if self.ast.avoid_insert.as_ref().unwrap().contains(&n) {
-                        return Err(
-                            self.mk_error(YaccParserErrorKind::DuplicateAvoidInsertDeclaration, i)
-                        );
+                    match self.ast.avoid_insert.as_ref().unwrap().get(&n) {
```

Comment:
May be better written as `self.ast.avoid_insert.entry(...)` (or one of the other combinators?) so you don't have to do a second `self.ast.avoid_insert.as_mut().unwrap().insert(n, span)`.